src/fuselage/CCPACSFuselage.cpp

```diff
@@ -372,6 +378,89 @@ TopoDS_Compound &CCPACSFuselage::GetGuideCurveWires()
     return guideCurves;
 }
 
+gp_Lin CCPACSFuselage::Intersection(gp_Pnt pRef, double angleRef)
+{
+    const gp_Ax1 xAxe(
+        pRef,
+        gp_Dir(1, 0,
+               0)); // to have a left-handed coordinates system for the intersection computation (see documentation)
+    const gp_Dir ZReference(0, 0, 1);
+    const gp_Dir angleDir = ZReference.Rotated(xAxe, angleRef + M_PI);
+
+    // build a line to position the intersection with the fuselage shape
+    gp_Lin line(pRef, angleDir);
+
+    // get the list of shape from the fuselage shape
+    TopExp_Explorer exp;
+    for (exp.Init(GetLoft()->Shape(), TopAbs_FACE); exp.More(); exp.Next()) {
+        IntCurvesFace_Intersector intersection(TopoDS::Face(exp.Current()), 0.1); // intersection builder
+        intersection.Perform(line, -std::numeric_limits<Standard_Real>::max(), // replace by std::numeric_limits<Standard_Real>::lowest() when C++11 available
+                             std::numeric_limits<Standard_Real>::max());
+        if (intersection.IsDone() && intersection.NbPnt() > 0) {
+            gp_Lin result(intersection.Pnt(1), line.Direction());
+            // return the line with the point on the fuselage as the origin, and the previous line's direction
+            return result;
+        }
+    }
+
+    throw std::logic_error("Error computing intersection line");
+}
+
+gp_Lin CCPACSFuselage::Intersection(const CCPACSFuselageStringerFramePosition& pos)
+{
+    const gp_Pnt pRef        = pos.GetRefPoint();
+    const double angleRefRad = (M_PI / 180.) * pos.GetReferenceAngle();
+    return Intersection(pRef, angleRefRad);
+}
+
+namespace
+{
+    TopoDS_Wire project(TopoDS_Shape wireOrEdge, BRepProj_Projection& proj)
```

Comment:
I am wondering, if this function could be generic function - i.e. not only for the fuselage.

Reply:
well, it's currently only used there. but you can pull it somewhere else if the need arises.